src/twisted/test/test_rebuild.py

```diff
@@ -80,6 +80,9 @@ def test_FileRebuild(self):
         self.assertEqual(i.a(), 'd')
         self.assertEqual(a.a(), 'b')
 
+    if _PYPY:
+        test_FileRebuild.skip = "Does not work on PYPY"
```

Comment:
See [Twisted ticket 9226](https://twistedmatrix.com/trac/ticket/9226) and [PyPy ticket](https://bitbucket.org/pypy/pypy/issues/2612/) this was a problem in PyPy which should be fixed in upstream and released in 5.8.1.

However looking closely at the way rebuild works it can be made much more efficient (whilst working everywhere now) by calling `gc.get_objects()` once rather than looping around `gc.get_referrers()`. I have a branch https://github.com/dansut/twisted/tree/9226-dansut-rebuild-get_objects that I was going to land as a PR after I'd looked at improving testing, and actually got some evidence on how much more efficient the new method is everywhere.